packages/mui-base/src/MenuUnstyled/MenuUnstyled.tsx

```diff
@@ -88,7 +89,12 @@ const MenuUnstyled = React.forwardRef(function MenuUnstyled<
     },
     className: classes.root,
     ownerState,
-  }) as MenuUnstyledRootSlotProps;
+  });
+
+  // Prevent passing `ownerState` to `PopperUnstyled`
+  if (!component && !slots.root && rootProps.ownerState) {
```

Comment:
This is not quite what I envisioned. The PopperUnstyled should ignore any incoming ownerState instead. This way any component that uses it won't need to worry about removing ownerState from props.

Reply:
Addressed it here: https://github.com/mui/material-ui/pull/36599/commits/7d3d705b2dfdc4da804c2e103a43f3be719c5966